mobile/components/feed/gnod-label.tsx

```diff
@@ -3,17 +3,29 @@ import { colors } from "@gno/styles/colors";
 import { MaterialCommunityIcons } from "@expo/vector-icons";
 import Text from "../text";
 import Icons from "../icons";
+import { Post } from "@gno/types";
 
 interface Props {
-  gnodsCount: number;
-  onPress?: () => void;
+  onGnod: (post: Post) => void;
+  post: Post;
   style?: StyleProp<TextStyle> | undefined;
 }
 
-function GnodLabel({ gnodsCount, style, onPress }: Props) {
+function GnodLabel({ post, style, onGnod }: Props) {
+  const isRepost = Boolean(post && post.parent_id > 0);
+  const gnodsCount = isRepost ? post.parent_post?.n_gnods : post.n_gnods;
+
+  const onPress = () => {
+    if (post.parent_post) {
```

Comment:
We have to separate "repost" from "reply". A repost has a parent AND a repost_user. A simple reply has a parent. This PR is about reposts, so I think we need to also check for repost_user, like here:
https://github.com/gnolang/dsocial/blob/0fa7d9a17928217668266d6d02ddd0c315d5a703/mobile/src/hooks/use-feed.ts#L63

Reply:
Nevermind. The code already checks.